### PR TITLE
Adds model validation to build

### PR DESCRIPTION
**Pull request by el-mapache**
* Invalid branch names will no longer persist to build records

### CODE REVIEW

File: test/api/unit/models/build.test.js [hunk 1]
```diff
@@ -1,160 +1,183 @@
...
-      const sitePromise = factory.site()
+        expect(build.token).to.equal('123abc');
+        done();
+      })
+      .catch(done);
+    });
+  });
+
+  describe('after create hook', () => {
+    // TODO: This doesnt test anything
```
Comment (jseppi) [inline test/api/unit/models/build.test.js]: ha, weird o_O

Looks like `SQS.sendBuildMessage` should be called with the newly created `buildId`. Want to add a quick expectation to check that?

File: test/api/unit/models/build.test.js [hunk 2]
```diff
@@ -1,160 +1,183 @@
...
+    it('should send a build new build message', (done) => {
+      const oldSendMessage = SQS.sqsClient.sendMessage;
+
+      SQS.sqsClient.sendMessage = () => {
+        SQS.sqsClient.sendMessage = oldSendMessage;
```
Comment (jseppi) [inline test/api/unit/models/build.test.js]: this behavior of restoring the old method should probably take place in `before()` and `after()` blocks

Comment (jseppi) [reply test/api/unit/models/build.test.js]: I know this wasn't part of this PR's change, but looks easy enough to fix this up and make it meaningful if you have time.

**Comment by el-mapache** [comment]
Added the expectation!

File: test/api/unit/models/build.test.js [hunk 3]
```diff
@@ -1,160 +1,194 @@
...
+
+    beforeEach(() => {
+      sendMessageStub = stub(SQS, 'sendBuildMessage');
+    });
+
+    afterEach(() => {
+      sendMessageStub.restore();
+    });
+
+    it('should send a build new build message', (done) => {
```
Comment (jseppi) [inline test/api/unit/models/build.test.js]: 👍 thanks!

**Pull request merged**